packages/mdc-ripple/util.ts

```diff
@@ -20,24 +20,30 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+import {MsElement} from '@material/dom/ponyfill';
+import {Point} from './adapter';
 
-/**
- * Stores result from supportsCssVariables to avoid redundant processing to detect CSS custom variable support.
- * @private {boolean|undefined}
- */
-let supportsCssVariables_;
+interface CssObject {
+  supports(prop: string, value?: string): boolean;
+}
+
+interface CssWindow extends Window {
+  CSS?: CssObject;
```

Comment:
TS has a type `CSS` which is equivalent to your `CssObject`, it's just not on the `Window` for some reason. You might be able to make a file of type fixes somewhere, like this:

```ts
declare interface Window { CSS: CSS; }
```

You could also add a fix for the `msMatchesSelector` thing, might be nicer than having to invent your own types and cast them everywhere

Reply:
ya we need to invent a place for these global types. Maybe the correct place is in mdc-dom package. I'm going to add it to the `index.ts` for now. We can decide for a better place as this PR evolves.

Reply:
Also I found that this works:
```
declare global {
  interface Window {
    CSS: CSS;
  }
}

```

globals seems like a last ditch effort. Is that a valid pattern? Or is there a better way of accomplishing this?